### DISCUSSION TITLE
Feature request - an environment variable like VERCEL_URL for fixed preview URLs

### DISCUSSION THREAD

**Discussion by mcky** [vercel/next.js#17332] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

I need to whitelist domains in a 3rd party service, but whitelisting `*.vercel.app` is too insecure.
I'd like to whitelist `*.mygroup.vercel.app` for preview urls like `project-git-my-branch.mygroup.vercel.app`

I also need to be able to pass the url to the third party service as part of the auth flow, but even when interacting with the project at `project-git-my-branch.mygroup.vercel.app` the `VERCEL_URL` param is still set to `project-***.vercel.app` 

The problem is I can only access `VERCEL_URL` which resolves to `project-***.vercel.app` and not the git branch based url

## Describe the solution you'd like

Another environment variable which represents the branch name version of the domain, e.g.
`VERCEL_STATIC_URL=project-git-my-branch.mygroup.vercel.app`

## Describe alternatives you've considered

Using the git env variables and slugifying the branch ourselves, but it seems too fragile

**Comment by jordn** [comment]
Same issue, need to whitelist the URL for our API. I don't trust `[project-name]-hash.vercel.app` as that would be insecure. 

I'd like all the preview urls to be on the team's subdomain

`*.myteam.vercel.app`
